cpp/bench/ann/src/common/dataset.h

```diff
@@ -270,31 +295,37 @@ Dataset<T>::~Dataset()
 {
   delete[] base_set_;
   delete[] query_set_;
-  if (d_base_set_) { RAFT_CUDA_TRY_NO_THROW(cudaFree(d_base_set_)); }
-  if (d_query_set_) { RAFT_CUDA_TRY_NO_THROW(cudaFree(d_query_set_)); }
+#ifndef CPU_ONLY
+  if (d_base_set_) { cudaFree(d_base_set_); }
+  if (d_query_set_) { cudaFree(d_query_set_); }
+#endif
 }
 
 template <typename T>
 const T* Dataset<T>::base_set_on_gpu() const
 {
+#ifndef CPU_ONLY
   if (!d_base_set_) {
     base_set();
-    RAFT_CUDA_TRY(cudaMalloc((void**)&d_base_set_, base_set_size_ * dim_ * sizeof(T)));
+    RAFT_CUDA_TRY(cudaMalloc((void**)&d_base_set_, base_set_size() * dim() * sizeof(T)));
```

Comment:
I wonder why we are not using RMM allocator here? (I know it was not changed in this PR, and should be a separate task to fix that).

Reply:
I think that was a straight file copy from the separate ann benchmarks project.

Reply:
Yep, this has been on the TODO but we rushed this out in 23.04 and haven't gotten back to this.

Reply:
Relevant issue here: https://github.com/rapidsai/raft/issues/1367. Item number 2